lib/AST/Expr.cpp

```diff
@@ -997,11 +1000,13 @@ swift::packSingleArgument(ASTContext &ctx, SourceLoc lParenLoc,
     return arg;
   }
 
-  // If we have no other arguments, represent the trailing closure as a
+  // If we have no other arguments, represent the a single trailing closure as a
   // parenthesized expression.
-  if (args.empty()) {
-    auto arg = new (ctx) ParenExpr(lParenLoc, trailingClosure, rParenLoc,
-                                   /*hasTrailingClosure=*/true);
+  if (args.empty() && trailingClosures.size() == 1) {
+    auto &trailingClosure = trailingClosures.front();
+    auto arg =
+        new (ctx) ParenExpr(lParenLoc, trailingClosure.ClosureExpr, rParenLoc,
+                            /*hasTrailingClosure=*/true);
```

Comment:
How do we distinguish `foo { 1 }` and `foo { fn: { 1 } }`? Also, this loses the parsed argument label.
```swift
func foo(bar: () -> Int) {}
foo { bang: { 12 } }
```
cannot be diagnosed.